ansible/roles/vm_set/tasks/start_k8s_vm.yml

```diff
@@ -0,0 +1,69 @@
+- name: Create directory for Ubuntu VM disks
+  file: path={{ k8s_root_path }}/{{ vm_name }} state=directory mode=0755
+
+- name: Check destination file existence
+  stat: path={{ disk_image }}
+  register: file_stat
+
+- name: "Generate cloud-config file for {{ vm_name }}"
+  template: 
+    src: templates/cloud-config.yml.j2
+    dest: "{{ k8s_root_path }}/{{ vm_name }}/cloud-config.yml"
+  when: not file_stat.stat.exists
```

Comment:
You could merge `stat` and `template` into just `template`.
ref: [https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_module.html](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_module.html)

Reply:
Merged